src/Fantomas.Core/CodePrinter.fs

```diff
@@ -308,6 +308,31 @@ let isSynExprLambdaOrIfThenElse (e: Expr) =
     | Expr.IfThenElse _ -> true
     | _ -> false
 
+let (|RecordOrAnonRecord|_|) (expr: Expr) =
+    match expr with
+    | Expr.Record record ->
+        Some
+            {| OpeningBrace = record.OpeningBrace
+               Extra = record.Extra
+               Fields = record.Fields
+               ClosingBrace = record.ClosingBrace
+               Range = record.Range
+               AnonRecord = None
```

Comment:
This exact field is a code smell on why these types cannot be combined.
There are subtle differences but in essence, one cannot be completely the other.
The code that needs to deal with this type will inevitably need to place `if/else`.

The anonymous record cannot have `inherit` and the regular record cannot have `struct`. That isn't very obvious from this refactor.